lib/HunkView.ml

```diff
@@ -31,44 +110,54 @@ let ui_hunk_summary (hunk : Patch.hunk) : Nottui.ui =
       at_symbols;
     ]
 
-let rec process_lines (mine_num : int) (their_num : int) (acc : Nottui.ui list)
-    = function
+let process_line (hunk : Patch.hunk) i line =
+  let line_number =
+    W.string
+      ~attr:Notty.A.(fg yellow)
+      (Printf.sprintf "%4d " (succ (hunk.Patch.mine_start + i)))
+  in
+  match line with
+  | Common s ->
+      let content = W.string s in
+      Ui.hcat [ line_number; content ]
+  | Change (_, word_diffs) ->
+      let word_uis =
+        List.map
+          (function
+            | WordDiff.WDeleted words ->
+                W.string
+                  ~attr:Notty.A.(bg red ++ fg white)
+                  (WordDiff.words_to_string words)
+            | WordDiff.WAdded words ->
+                W.string
+                  ~attr:Notty.A.(bg green ++ fg black)
+                  (WordDiff.words_to_string words)
+            | WordDiff.WEqual words -> W.string (WordDiff.words_to_string words))
+          word_diffs
+      in
+      let space = W.string " " in
+      let content =
+        Ui.hcat
+          (List.fold_left (fun acc ui -> ui :: space :: acc) [] word_uis
+          |> List.rev)
+      in
+      Ui.hcat [ line_number; content ]
```

Comment:
I find this rendering to be confusing. For a first version of the feature we definitely want to render the diffs as before, with removed and added lines. The word diff information should be used to highlight words within these lines.